embassy-rp/src/uart/buffered.rs

```diff
@@ -367,40 +305,31 @@ pub(crate) unsafe fn on_interrupt<T: Instance>(_: *mut ()) {
     let s = T::state();
 
     unsafe {
-        // RX
-
         let ris = r.uartris().read();
-        // Clear interrupt flags
-        r.uarticr().write(|w| {
-            w.set_rxic(true);
-            w.set_rtic(true);
-        });
+        let mut mis = r.uartimsc().read();
 
-        if ris.peris() {
-            warn!("Parity error");
-            r.uarticr().write(|w| {
-                w.set_peic(true);
-            });
-        }
+        // Errors
         if ris.feris() {
             warn!("Framing error");
-            r.uarticr().write(|w| {
-                w.set_feic(true);
-            });
+        }
+        if ris.peris() {
+            warn!("Parity error");
         }
         if ris.beris() {
             warn!("Break error");
-            r.uarticr().write(|w| {
-                w.set_beic(true);
-            });
         }
         if ris.oeris() {
             warn!("Overrun error");
-            r.uarticr().write(|w| {
-                w.set_oeic(true);
-            });
         }
+        // Clear any error flags
+        r.uarticr().write(|w| {
```

Comment:
if an error happens between reading `uartimsc` and this write, it'll get lost.

Reply:
fixed by only clearing the pending flags